<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from mybot.xacro                    | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="mybot" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <static>false </static>
  <!-- ======================definisi warna===================================== -->
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <!-- ======================eof definisi warna===================================== -->
  <!-- == eof makro variable==== -->
  <!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% -->
  <!-- 1. =====================world definition ======================================== -->
  <!-- <link name="world"/> -->
  <!-- 1. == eof world definition ===== -->
  <!-- 2. ===================== base definition ======================================== -->
  <!-- <joint name="joint0" type="fixed"> -->
  <!--   <origin xyz=" ${p_base/2} ${l_base/2} ${t_base}" rpy="0 0 0"/> -->
  <!--   <axis xyz="0 0 1"/> -->
  <!--   <parent link="world"/> -->
  <!--   <child link="base"/> -->
  <!-- </joint> -->
  <link name="base">
    <visual>
      <geometry>
        <box size="0.18 0.08 0.04"/>
      </geometry>
      <!-- <origin xyz=" ${p_base/2} ${l_base/2} ${t_base}" rpy="0 0 0"/> -->
    </visual>
    <collision>
      <geometry>
        <box size="0.18 0.08 0.04"/>
      </geometry>
      <!-- <origin xyz=" ${p_base/2} ${l_base/2} ${t_base}" rpy="0 0 0"/> -->
    </collision>
    <inertial>
      <mass value="1"/>
      <inertia ixx="0.0006666664" ixy="0.0" ixz="0.0" iyy="0.0028333322" iyz="0.0" izz="0.00323333204"/>
    </inertial>
  </link>
  <!-- <gazebo reference="base"> -->
  <!--   <mu1>0.2</mu1> -->
  <!--   <mu2>0.2</mu2> -->
  <!--   <material>Gazebo/Blue</material> -->
  <!--   <turnGravityOff>false</turnGravityOff> -->
  <!-- </gazebo> -->
  <!-- 2. oef base definition -->
  <!-- 3. ===================== rfw definition ======================================== -->
  <joint name="joint1" type="continuous">
    <origin rpy="1.57079632679 0 0" xyz="0.04 0.052 0"/>
    <axis xyz="0 0 1"/>
    <parent link="base"/>
    <child link="right_front_wheel"/>
  </joint>
  <link name="right_front_wheel">
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
      <!-- <origin xyz="0.04 ${((l_base/2)+(wheel_length/2)+wheel_base_dist)} 0" rpy="${M_PI/2} 0 0"/> -->
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.7"/>
      <inertia ixx="0.000180833333333" ixy="0" ixz="0" iyy="0.000180833333333" iyz="0" izz="0.000315"/>
    </inertial>
  </link>
  <!-- <gazebo reference="right_front_wheel"> -->
  <!--   <mu1>0.2</mu1> -->
  <!--   <mu2>0.2</mu2> -->
  <!--   <material>Gazebo/Red</material> -->
  <!--   <turnGravityOff>false</turnGravityOff> -->
  <!-- </gazebo> -->
  <!-- 4. eof rbw definition === -->
  <!-- 5. ===================== lfw definition ======================================== -->
  <joint name="joint2" type="continuous">
    <origin rpy="1.57079632679 0 0" xyz="0.04 -0.052 0"/>
    <axis xyz="0 0 1"/>
    <parent link="base"/>
    <child link="left_front_wheel"/>
  </joint>
  <link name="left_front_wheel">
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
      <!-- <origin xyz="0.04 ${0-((l_base/2)+(wheel_length/2)+wheel_base_dist)} 0" rpy="${M_PI/2} 0 0"/> -->
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.02" radius="0.03"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.7"/>
      <inertia ixx="0.000180833333333" ixy="0" ixz="0" iyy="0.000180833333333" iyz="0" izz="0.000315"/>
    </inertial>
  </link>
  <!-- <gazebo reference="left_front_wheel"> -->
  <!--   <mu1>0.2</mu1> -->
  <!--   <mu2>0.2</mu2> -->
  <!--   <material>Gazebo/Green</material> -->
  <!--   <turnGravityOff>false</turnGravityOff> -->
  <!-- </gazebo> -->
  <!-- 5.  eof lbw definition= -->
  <!-- 6. ===================== rbw definition ======================================== -->
  <!-- <joint name="joint3" type="continuous"> -->
  <!--   <origin xyz="-0.04 ${((l_base/2)+(wheel_length/2)+wheel_base_dist)} 0" rpy="${M_PI/2} 0 0"/> -->
  <!--   <axis xyz="0 0 1"/> -->
  <!--   <parent link="base"/> -->
  <!--   <child link="right_back_wheel"/> -->
  <!-- </joint> -->
  <!-- <link name="right_back_wheel"> -->
  <!--   <visual> -->
  <!--     <geometry> -->
  <!--       <cylinder length="${wheel_length}" radius="${wheel_diameter}"/> -->
  <!--     </geometry> -->
  <!--   </visual> -->
  <!--   <collision> -->
  <!--     <geometry> -->
  <!--       <cylinder length="${wheel_length}" radius="${wheel_diameter}"/> -->
  <!--     </geometry> -->
  <!--   </collision> -->
  <!--   <inertial> -->
  <!--     <mass value="${wheel_mass}"/> -->
  <!--     <cylinder_inertia m="${wheel_mass}" r="${wheel_diameter}" h="${wheel_length}"/> -->
  <!--   </inertial> -->
  <!-- </link> -->
  <!-- <gazebo reference="right_back_wheel"> -->
  <!--   <mu1>0.2</mu1> -->
  <!--   <mu2>0.2</mu2> -->
  <!--   <material>Gazebo/Red</material> -->
  <!--   <turnGravityOff>false</turnGravityOff> -->
  <!-- </gazebo> -->
  <!-- 6. eof rbw definition === -->
  <!-- 5. ===================== lbw definition ======================================== -->
  <!-- <joint name="joint4" type="continuous"> -->
  <!--   <origin xyz="-0.04 ${0-((l_base/2)+(wheel_length/2)+wheel_base_dist)} 0" rpy="${M_PI/2} 0 0"/> -->
  <!--   <axis xyz="0 0 1"/> -->
  <!--   <parent link="base"/> -->
  <!--   <child link="left_back_wheel"/> -->
  <!-- </joint> -->
  <!-- <link name="left_back_wheel"> -->
  <!--   <visual> -->
  <!--     <geometry> -->
  <!--       <cylinder length="${wheel_length}" radius="${wheel_diameter}"/> -->
  <!--     </geometry> -->
  <!--   </visual> -->
  <!--   <collision> -->
  <!--     <geometry> -->
  <!--       <cylinder length="${wheel_length}" radius="${wheel_diameter}"/> -->
  <!--     </geometry> -->
  <!--   </collision> -->
  <!--   <inertial> -->
  <!--     <mass value="${wheel_mass}"/> -->
  <!--     <cylinder_inertia m="${wheel_mass}" r="${wheel_diameter}" h="${wheel_length}"/> -->
  <!--   </inertial> -->
  <!-- </link> -->
  <!-- <gazebo reference="left_back_wheel"> -->
  <!--   <mu1>0.2</mu1> -->
  <!--   <mu2>0.2</mu2> -->
  <!--   <material>Gazebo/Green</material> -->
  <!--   <turnGravityOff>false</turnGravityOff> -->
  <!-- </gazebo> -->
  <!-- 7.  eof lbw definition= -->
  <!-- <gazebo> -->
  <!--   <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so"> -->
  <!--     <robotNamespace>/mybot</robotNamespace> -->
  <!--   </plugin> -->
  <!-- </gazebo> -->
  <!-- <gazebo> -->
  <!--   <plugin name="differential_drive_controller" filename="libgazebo_ros_diff_drive.so"> -->
  <!--     <alwaysOn>true</alwaysOn> -->
  <!--     <updateRate>100</updateRate> -->
  <!--     <leftJoint>left_wheel_hinge</leftJoint> -->
  <!--     <rightJoint>right_wheel_hinge</rightJoint> -->
  <!--     <wheelSeparation>${l_base}</wheelSeparation> -->
  <!--     <wheelDiameter>${2*wheel_diameter}</wheelDiameter> -->
  <!--     <torque>20</torque> -->
  <!--     <commandTopic>mybot/cmd_vel</commandTopic> -->
  <!--     <odometryTopic>mybot/odom_diffdrive</odometryTopic> -->
  <!--     <odometryFrame>odom</odometryFrame> -->
  <!--     <robotBaseFrame>footprint</robotBaseFrame> -->
  <!--     <publishWheelTF>true</publishWheelTF> -->
  <!--     <publishWheelJointState>true</publishWheelJointState> -->
  <!--     <legacyMode>false</legacyMode> -->
  <!--     <wheelAcceleration>10</wheelAcceleration> -->
  <!--     <wheelTorque>5</wheelTorque> -->
  <!--     <rosDebugLevel>na</rosDebugLevel> -->
  <!--     <odometrysource>1</odometrysource> -->
  <!--   </plugin> -->
  <!-- </gazebo> -->
  <!-- <transmission name="tran1"> -->
  <!--   <type>transmission_interface/SimpleTransmission</type> -->
  <!--   <joint name="joint1"> -->
  <!--     <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface> -->
  <!--   </joint> -->
  <!--   <actuator name="motor1"> -->
  <!--     <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface> -->
  <!--     <mechanicalReduction>10</mechanicalReduction> -->
  <!--   </actuator> -->
  <!-- </transmission> -->
  <!-- <transmission name="tran2"> -->
  <!--   <type>transmission_interface/SimpleTransmission</type> -->
  <!--   <joint name="joint2"> -->
  <!--     <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface> -->
  <!--   </joint> -->
  <!--   <actuator name="motor2"> -->
  <!--     <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface> -->
  <!--     <mechanicalReduction>10</mechanicalReduction> -->
  <!--   </actuator> -->
  <!-- </transmission> -->
</robot>

